metrics_api/lib/opentelemetry/internal/proxy_instrument.rb

```diff
@@ -8,19 +8,22 @@ module OpenTelemetry
   module Internal
     # @api private
     class ProxyInstrument
-      def initialize(kind, name, unit, desc, callable)
+      def initialize(kind, name, unit, desc, callable, advice = nil)
```

Comment:
Also, `advice` is shorter than `advisory_parameters`, so that's why I chose that word. However, it's not the latest terminology used in the spec. I'm happy to change to `advisory_parameters` if that's preferred. Perhaps some documentation would clarify things.